program-runtime/src/invoke_context.rs

```diff
@@ -1135,6 +1135,7 @@ pub fn mock_process_instruction(
     instruction_data: &[u8],
     transaction_accounts: Vec<TransactionAccount>,
     instruction_accounts: Vec<AccountMeta>,
+    sysvar_cache_override: Option<&SysvarCache>,
```

Comment:
@jstarry Why not just use the existing functionality in `InvokeContext::new_mock()` to fill / override the sysvar cache?
Adding the `Rent` sysvar to the `transaction_accounts` (even if it is not used by any of the `instruction_accounts`) should also work I think.

https://github.com/solana-labs/solana/blob/75c6747b22950823a7f29f854f28f630b2059d40/program-runtime/src/invoke_context.rs#L242

Reply:
Because I don't think it makes any sense to require the transaction accounts to list a sysvar account when they can be fetched from the cache. The mock invoke context shouldn't be filling its sysvar cache like that, it doesn't make much sense

Reply:
I agree that it is strange to pass sysvars in the transaction accounts. However, it comes from sysvars being accounts in the first place. Is that what bothers you or is there anything else?

The sysvar cache has to be populated somehow and until we remove the need of sysvars being accounts, I think it is acceptable. What bothers me is that we only use this special parameter `sysvar_cache_override` in one place atm, and it could do without.

Reply:
> I agree that it is strange to pass sysvars in the transaction accounts. However, it comes from sysvars being accounts in the first place. Is that what bothers you or is there anything else?

We use the mock invoke context for testing native programs which directly use the sysvar cache. I think that the mock invoke context should allow for setting up a mocked sysvar cache. When sysvar accounts are loaded as accounts, we either use them directly or override them with the sysvar cache. In either case, it doesn't make sense to setup the mock context's sysvar cache with transaction accounts.